src/python/pants/jvm/target_types.py

```diff
@@ -118,28 +126,31 @@ class JvmArtifact(Target):
         JvmArtifactJarSourceField,
     )
     help = (
-        "Represents a third-party JVM artifact as identified by its Maven-compatible coordinate, "
-        "that is, its `group`, `artifact`, and `version` components."
+        "A third-party JVM artifact, as identified by its Maven-compatible coordinate.\n\n"
+        "That is, an artifact identified by its `group`, `artifact`, and `version` components."
     )
 
 
 class JvmCompatibleResolveNamesField(StringSequenceField):
     alias = "compatible_resolves"
     required = False
     help = (
-        "The set of resolve names that this target is compatible with. Any targets which depend on "
-        "one another must have at least one compatible resolve in common. Which resolves are actually "
-        "used in a build is calculated based on a target's dependees."
+        "The set of resolve names that this target is compatible with.\n\n"
+        "The name must be defined as one of the resolves in `[jvm].resolves`.\n\n"
+        "Any targets which depend on one another must have at least one compatible resolve in "
+        "common. Which resolves are actually used in a build is calculated based on a target's "
+        "dependees."
     )
 
 
 class JvmResolveNameField(StringField):
     alias = "resolve"
     required = False
     help = (
-        "The name of the resolve to use when building this target. The name must be defined as "
-        "one of the resolves in `--jvm-resolves`. If not supplied, the default resolve will be "
-        "used, otherwise, one resolve that is compatible with all dependency targets will be used."
+        "The name of the resolve to use when building this target.\n\n"
+        "The name must be defined as one of the resolves in `[jvm].resolves`.\n\n"
+        "If not supplied, the default resolve will be used. Otherwise, one resolve that is "
+        "compatible with all dependency targets will be used."
```

Comment:
That's not how I envisioned things working with Python. The idea was that this field will force you to use a particular resolve, but we'll first validate that all dependees are compatible with what's selected.

Reply:
I know this statement was already here, but it strikes me as a bit odd in the context of #12742. The implication of #12742 is that we'll _always_ have a default resolve, and so the "otherwise, one resolve" bit isn't/won't be necessary.

> but we'll first validate that all dependees are compatible with what's selected.

I'm not sure that this is strictly necessary, because building any `dependees` would do that validation. It shouldn't be necessary to walk into dependees, afaict.

Reply:
As it stands, we don't always have a default resolve. But that's one of the changes I have planned for this week.